### PR TITLE
TextField and PageController updated

### PR DESCRIPTION
**Pull request by Azzeccagarbugli**
So basically, as you suggested during the wonderful [stream](https://www.twitch.tv/videos/1592863677), I fixed the following:

- `Viewport fraction` of the `PageController` 
- Improved `TextField` from my previous submission
- `AnnotedRegion` for the status bar brightness
- Clear `TextField` button just when the `TextField` itself is not empty

Enjoy 🎉

### CODE REVIEW

**Comment by lukepighetti** [comment]
Hey! Can you post up a screenshot to go with your PR? Will make it easier to review. Thanks

**Comment by Azzeccagarbugli** [comment]
> Hey! Can you post up a screenshot to go with your PR? Thanks

Sure ✅

| Full screen  | Keyboard |
| ------------- | ------------- |
| Simulator Screen Shot - iPhone 14 Pro - 2022-09-17 at 22 17 47 | Simulator Screen Shot - iPhone 14 Pro - 2022-09-17 at 22 18 04 |

**Comment by lukepighetti** [comment]
Let's

- revert the back button style, 
- revert the text field text style, 
- re-center the text field,
- change the clear button to a basic material [clear](https://api.flutter.dev/flutter/material/Icons/clear-constant.html) icon 
- and make sure the text is centered

**Comment by Azzeccagarbugli** [comment]
> * change the clear button to a basic material [clear](https://api.flutter.dev/flutter/material/Icons/clear-constant.html) icon

In this way, both the icons of the `go-back` action and the `clear-text` one will have the same icon though 

> * and make sure the text is centered

I really can't find a proper way to do this, I even tried a hack with a `Stack` widget but doesn't feel like the right approach

**Comment by lukepighetti** [comment]
We can leave out the clear button for now then, that should make it a lot easier

**Comment by Azzeccagarbugli** [comment]
> We can leave out the clear button for now then, that should make it a lot easier

Do you mean leaving it out from the container that holds the `TextField` or completely removing it from the UI?

**Comment by lukepighetti** [comment]
> > We can leave out the clear button for now then, that should make it a lot easier
> 
> Do you mean leaving it out of the container that holds the `TextField` or completely removing it from the UI?

If we can't keep everything nicely centered, let's remove the clear button feature completely

**Comment by Azzeccagarbugli** [comment]
The last [commit](https://github.com/lukepighetti/pocketbase_flutter_app/pull/3/commits/2a3cb83300d78b3641600ad5f0a5bed0844953fb) should be fine now. Waiting for your feedback!

File: pb_app/lib/submission_form.dart [hunk 1]
```diff
@@ -24,80 +25,90 @@ class _SubmissionFormScreenState extends State<SubmissionFormScreen> {
...
+      child: Scaffold(
+        appBar: PreferredSize(
+          preferredSize: const Size.fromHeight(60),
+          child: SafeArea(
             child: Row(
               crossAxisAlignment: CrossAxisAlignment.center,
               children: [
                 IconButton(
-                  icon: const Icon(Icons.close),
+                  icon: const Icon(Icons.close_rounded),
```
Comment (lukepighetti) [inline pb_app/lib/submission_form.dart]:
```suggestion
                  icon: const Icon(Icons.close),
```

File: pb_app/lib/submission_form.dart [hunk 2]
```diff
@@ -24,80 +25,90 @@ class _SubmissionFormScreenState extends State<SubmissionFormScreen> {
...
+                        decoration: BoxDecoration(
+                          borderRadius: platformAwareBorderRadius(10),
+                          color: Colors.black12,
+                        ),
+                        child: TextField(
+                          autofocus: true,
+                          focusNode: _nameFocusNode,
+                          textAlign: TextAlign.center,
+                          style: const TextStyle(fontSize: 20),
+                          cursorColor: Colors.black26,
```
Comment (lukepighetti) [inline pb_app/lib/submission_form.dart]:
```suggestion
```

Comment (lukepighetti) [reply pb_app/lib/submission_form.dart]: Let's not override the theme cursor color locally :)

File: pb_app/lib/submission_form.dart [hunk 3]
```diff
@@ -24,80 +25,90 @@ class _SubmissionFormScreenState extends State<SubmissionFormScreen> {
...
-                /// Learn the rules like a scientist
-                /// so you can break them like an artist.
                 const Visibility(
                   visible: false,
                   maintainSize: true,
                   maintainAnimation: true,
                   maintainState: true,
                   child: IconButton(
-                    icon: Icon(Icons.close),
+                    icon: Icon(Icons.close_rounded),
```
Comment (lukepighetti) [inline pb_app/lib/submission_form.dart]:
```suggestion
                    icon: Icon(Icons.close),
```

File: pb_app/lib/submission_form.dart [hunk 4]
```diff
@@ -24,80 +25,90 @@ class _SubmissionFormScreenState extends State<SubmissionFormScreen> {
...
-          style: ElevatedButton.styleFrom(
-            shape: SmoothRectangleBorder(
-              borderRadius: platformAwareBorderRadius(10),
+        body: PageView.builder(
+          controller: _pageController,
+          itemCount: 2,
+          physics: const BouncingScrollPhysics(),
+          itemBuilder: (context, index) => index == 0
+              ? const _Card('Home Screen')
+              : const _Card('Lock Screen'),
```
Comment (lukepighetti) [inline pb_app/lib/submission_form.dart]:
```suggestion
         body: PageView(
           children: const [
            _Card('Home Screen'),
            _Card('Lock Screen'),
          ],
```

We won't see a performance benefit using PageView.builder here, so the added complexity is not worth it

File: pb_app/lib/submission_form.dart [hunk 5]
```diff
@@ -112,33 +123,22 @@ class _Card extends StatelessWidget {
...
+        vertical: 25,
+        horizontal: 5,
+      ),
       color: Colors.black38,
       shape: platformAwareShape(50),
       child: InkWell(
-        onTap: () {},
+        onTap: () {
+          // TODO: InkWell not working properly into stack (https://github.com/flutter/flutter/issues/104519)
+          // debugPrint('I have been clicked even if the ripple isn\'t showing');
```
Comment (lukepighetti) [inline pb_app/lib/submission_form.dart]:
```suggestion
```

**Review by lukepighetti — CHANGES_REQUESTED** [review]
Couple more changes requested, thanks!

**Comment by lukepighetti** [comment]
Also, let's get this up to date with main branch so we can get ci checks on it

**Comment by Azzeccagarbugli** [comment]
Thank a lot for the code review, hopefully, the process of the next PR is going to be more seamless.

Would be pretty helpful to have a project board or a bunch of issues so that we could work **specifically** on some features.

**Comment by lukepighetti** [comment]
Yes! Please bare with me as I make it easier for folks to contribute.

Had one more change request (my mistake!) And the files do need to get reformatted.

Thank you

**Comment by lukepighetti** [comment]
You can also merge in main branch and it should auto format and commit for you due to https://github.com/lukepighetti/pocketbase_flutter_app/commit/0e2a4fef84c3a7907a7b62c69cd0515e542a07ac

**Comment by Azzeccagarbugli** [comment]
> Yes! Please bare with me as I make it easier for folks to contribute.
> 
> Had one more change request (my mistake!) And the files do need to get reformatted.
> 
> Thank you

The code review is marked as resolved so I don't understand which other request I'm missing 😅

**Comment by lukepighetti** [comment]
Woops! I must have forgotten to submit the review on my office computer. The page controller needs to be hooked up again.

**Comment by Azzeccagarbugli** [comment]
> Woops! I must have forgotten to submit the review on my office computer. The page controller needs to be hooked up again.

Done it ✅

File: pb_app/lib/submission_form.dart [hunk 6]
```diff
@@ -112,33 +121,21 @@ class _Card extends StatelessWidget {
...
+      margin: const EdgeInsets.symmetric(
+        vertical: 25,
+        horizontal: 5,
+      ),
       color: Colors.black38,
       shape: platformAwareShape(50),
       child: InkWell(
-        onTap: () {},
+        onTap: () {
+          // TODO: InkWell not working properly into stack (https://github.com/flutter/flutter/issues/104519)
```
Comment (lukepighetti) [inline pb_app/lib/submission_form.dart]: This issue was resolved in https://github.com/lukepighetti/pocketbase_flutter_app/commit/e374c182affd4a2debd5b4b0a1647de493d78685 so the comment should be removed

File: pb_app/lib/submission_form.dart [hunk 7]
```diff
@@ -112,33 +121,21 @@ class _Card extends StatelessWidget {
...
-                ),
-                child: const Center(
-                  child: Icon(
-                    Icons.add,
-                    color: Colors.white,
-                    size: 28,
-                  ),
-                ),
-              ),
-            ),
```
Comment (lukepighetti) [inline pb_app/lib/submission_form.dart]: can we revert these changes? they are having no effect

File: pb_app/lib/submission_form.dart [hunk 8]
```diff
@@ -155,7 +152,23 @@ class _Card extends StatelessWidget {
...
+                ),
+                child: const Center(
+                  child: Icon(
+                    Icons.add,
+                    color: Colors.white,
+                    size: 28,
+                  ),
+                ),
+              ),
+            ),
```
Comment (lukepighetti) [inline pb_app/lib/submission_form.dart]: can we revert these changes? they are having no effect

**Review by lukepighetti — CHANGES_REQUESTED** [review]
Found a couple more things, I think we're nearly there.

As they say, 'first one through the wall gets bloody.'

**Comment by lukepighetti** [comment]
Thank you!

**Pull request merged**